src/App/Tracker/Controller/Hooks/Listeners/JoomlacmsIssuesListener.php

```diff
@@ -38,109 +39,35 @@ public function onIssueAfterCreate(Event $event)
-	 * @param   Github  $github    Github object
-	 * @param   Logger  $logger    Logger object
-	 * @param   object  $project   Object containing project data
+	 * @param   object       $hookData  Hook data payload
+	 * @param   Github       $github    Github object
+	 * @param   Logger       $logger    Logger object
+	 * @param   object       $project   Object containing project data
+	 * @param   IssuesTable  $table     Table object
 	 *
 	 * @return  void
 	 *
 	 * @since   1.0
 	 */
-	protected function addCodelabel($hookData, Github $github, Logger $logger, $project)
+	protected function checkNoCodelabel($hookData, Github $github, Logger $logger, $project, IssuesTable $table)
 	{
 		// Set some data
-		$codeLabel    = 'No Code Attached Yet';
-		$addLabels    = array();
-		$codeLabelSet = false;
+		$label      = 'No Code Attached Yet';
+		$labels     = array();
+		$labelIsSet = $this->checkLabel($hookData, Github $github, Logger $logger, $project, IssuesTable $table, $label);
 
-		// Get the labels for the issue
-		try
+		if ($labelIsSet == false)
 		{
-			$labels = $github->issues->get($project->gh_user, $project->gh_project, $hookData->issue->number)->labels;
-		}
-		catch (\DomainException $e)
-		{
-			$logger->error(
-				sprintf(
-					'Error retrieving labels for GitHub item %s/%s #%d - %s',
-					$project->gh_user,
-					$project->gh_project,
-					$hookData->issue->number,
-					$e->getMessage()
-				)
-			);
-
-			return;
-		}
-
-		// Check if the label is present only if there are already labels attached to the item
-		if (count($labels) > 0)
-		{
-			foreach ($labels as $label)
-			{
-				if (!$codeLabelSet && $label->name == $codeLabel)
-				{
-					$logger->info(
-						sprintf(
-							'GitHub item %s/%s #%d already has the %s label.',
-							$project->gh_user,
-							$project->gh_project,
-							$hookData->issue->number,
-							$codeLabel
-						)
-					);
-
-					$codeLabelSet = true;
-				}
-			}
-		}
-
-		// Add the label if it isn't already set
-		if (!$codeLabelSet)
-		{
-			$addLabels[] = $codeLabel;
-		}
-
-		// Only try to add labels if the array isn't empty
-		if (!empty($addLabels))
-		{
-			try
-			{
-				$github->issues->labels->add(
-					$project->gh_user, $project->gh_project, $hookData->issue->number, $addLabels
-				);
-
-				// Post the new label on the object
-				$logger->info(
-					sprintf(
-						'Added %s labels to %s/%s #%d',
-						count($addLabels),
-						$project->gh_user,
-						$project->gh_project,
-						$hookData->issue->number
-					)
-				);
-			}
-			catch (\DomainException $e)
-			{
-				$logger->error(
-					sprintf(
-						'Error adding labels to GitHub issue %s/%s #%d - %s',
-						$project->gh_user,
-						$project->gh_project,
-						$hookData->issue->number,
-						$e->getMessage()
-					)
-				);
-			}
+			// Add the label as it isn't already set
+			$labels[] = $label;
+			$this->addLabels($hookData, Github $github, Logger $logger, $project, IssuesTable $table, $labels);
```

Comment:
You should remove type hinting here. And also do not forget to remove `$table` as it is not used.